src/meta/src/hummock/manager/transaction.rs

```diff
@@ -149,14 +150,23 @@ impl<'a> HummockVersionTransaction<'a> {
 
         // Append SSTs to a new version.
         for (compaction_group_id, inserted_table_infos) in commit_sstables {
+            // There is no necessity to invoke may_bump_max_sub_level_id here, because
+            // - There's at most one IntraLevel delta for each compaction group in one pre_commit_epoch's delta.
+            // - No other delta type will be present.
+            // - may_bump_max_sub_level_id will be invoked once later for this delta in apply_version_delta.
+            let l0_sub_level_id = new_version_delta
+                .latest_version()
+                .max_sub_level_id
+                .map(|id| id + 1)
+                .unwrap_or(FIRST_SUB_LEVEL_ID);
```

Comment:
For extra safety, I suggest we fill in `max_sub_level_id` using `Epoch::now()` if it is None on meta startup (in load_meta_store_state_impl) after all existing deltas are applied. In this case, we can make sure `max_sub_level_id` is larger than any previously generated sub level id.

Reply:
Yes this should be the correct way. 
Previously I didn't consider the delta replay for backward-compatibility. Backward compatibility shouldn't be done during pb deserialization.

Reply:
Refactored:
- In-mem max_sub_level_id's type is u64 instead of Option<u64> now.
- backward compatibility is ensured during `load_meta_store_state_impl`.